lib/auth/auth_with_roles.go

```diff
@@ -4505,12 +4519,35 @@ func (a *ServerWithRoles) ReplaceRemoteLocks(ctx context.Context, clusterName st
 // channel if one is encountered. Otherwise the event channel is closed when the stream ends.
 // The event channel is not closed on error to prevent race conditions in downstream select statements.
 func (a *ServerWithRoles) StreamSessionEvents(ctx context.Context, sessionID session.ID, startIndex int64) (chan apievents.AuditEvent, chan error) {
-	if err := a.actionForKindSession(apidefaults.Namespace, types.VerbList, sessionID); err != nil {
+	createErrorChannel := func(err error) (chan apievents.AuditEvent, chan error) {
 		c, e := make(chan apievents.AuditEvent), make(chan error, 1)
 		e <- trace.Wrap(err)
 		return c, e
 	}
 
+	if err := a.actionForKindSession(apidefaults.Namespace, types.VerbList, sessionID); err != nil {
+		return createErrorChannel(err)
+	}
+
+	// StreamSessionEvents can be called internally, and when that happens we don't want to emit an event.
+	_, isBuiltinRole := a.context.Identity.(BuiltinRole)
+	shouldEmitAuditEvent := !isBuiltinRole
```

Comment:
If we end up with `Nop` or some other weird builtin role we probably want to log the access (not that we should ever get to this point like that).

```suggestion
	shouldEmitAuditEvent := true
	if role, ok := a.context.Identity.(BuiltinRole); ok {
		shouldEmitAuditEvent = !role.IsServer()
	}
```

Reply:
(or `if role.IsServer() { shouldEmitAuditEvent = false }` which is a bit clearer)

Reply:
Nice one, I've changed this